minio/minio
# Lookup existing secrets

## Description

Use the lookup method provided by Helm to find out whether the previously generated secrets already exists in the K8s where the current Release is located before generating a random key. Avoid generating new useless random keys on every upgrade.

## Motivation and Context

Th Random secrets are an easy way to keep the production environment secure. Unfortunately when you try to upgrade the release, the process of generating the random password will be performed again and the new random password will be written to Secrets. But at this time the application has completed the initialization process, the new password will not be adopted, and it becomes an orphan password. At this point, it becomes difficult for you to retrieve the original random password.

## How to test this PR?

Create a Release with the new Chart, which will generate a new random password for the root user and save it in a secret. Then try to upgrade the release and check again that the random key in the secret has not changed.

## Types of changes

## Checklist:
